src/background.js

```diff
@@ -158,31 +158,36 @@ function validateSignature(content, signature, pageOptions, pubkey) {
   });
 }
 
+// Returns when first promise returns true
+function promise_any(promises) {
+  return new Promise((resolve,reject) => {
+    Promise.all(promises.map(promise => promise.then(value => {
+      if (value)
+        resolve(value);
+      return value;
+    }).catch( error => {
+      reject(error);
+    }))).then(values =>  {
+      if (values.every(x => x == false))
+        resolve(false)
```

Comment:
Okay, this resolves promises with `false`. So it is only supported for promises that return boolean values…